scripts/unicode_lister.rb

```diff
@@ -163,18 +164,18 @@ def calculate_ranges(matching_codes)
     end
   end
     .map { |range| range.size == 1 ? range.first : range}
-    .join(", ")
 end
 
 count = 0
-File.open(OutputFilename, 'w') do |f|
+store = PStore.new("db/#{RegexpExamples::UnicodeCharRanges::STORE_FILENAME}")
+store.transaction do
   NamedGroups.each do |name|
   count += 1
     matching_codes = (0..55295).lazy.select { |x| /\p{#{name}}/ =~ eval("?\\u{#{x.to_s(16)}}") }.first(128)
-    f.puts "'#{name.downcase}' => ranges_to_unicode(#{calculate_ranges(matching_codes)}),"
+    store[name.downcase] = calculate_ranges(matching_codes)
```

Comment:
Good point about performance - running that script on my machine gives:

       user     system      total        real
yaml 45.020000   0.140000  45.160000 ( 45.167915)
pstore  1.680000   0.050000   1.730000 (  1.725712)

Performance is unlikely to be an issue in "real use" of the system, but could become significant in a larger test suite. For example, if Onigmo implements support for extended bracket character classes (in Perl: http://perldoc.perl.org/perlrecharclass.html#Extended-Bracketed-Character-Classes), then this gem might need additional features+tests added for ruby 2.3+

So the question is really: Do we want to maximise the gem performance (with a pstore), or provide a human-readable file diff between ruby versions?

I guess even though the latter would be nice to have, it should not be the responsibility of this gem to effectively provide (what should be) ruby documentation. Perhaps I could re-run the script in YAML-mode and save the result in a blog post/something.

tl;dr: PStore is fine, ruby should document this stuff not me :stuck_out_tongue: